File: 백준/Gold/18291. 비요뜨의 징검다리 건너기/비요뜨의 징검다리 건너기.cc
#include <iostream>

using namespace std;
constexpr int MOD = 1e9 + 7;

int64_t Pow(int64_t Base, int64_t Exp)
{
    if(Exp == 1)
    {
        return Base;
    }
    
    int64_t NxtBase = Pow(Base, Exp / 2);
    if(Exp % 2 == 0)
    {
        return (NxtBase * NxtBase) % MOD;
    }
    else
    {
        return (NxtBase * NxtBase * Base) % MOD;
    }
}

int main()
{
    cin.tie(nullptr)->sync_with_stdio(false);
    
    int T; cin >> T;
    while(T--)
    {
        int N; cin >> N;
        cout << (N <= 2 ? 1 : Pow(2, N - 2)) << "\n";
    }
    
    return 0;
}

File: 백준/Gold/18291. 비요뜨의 징검다리 건너기/비요뜨의 징검다리 건너기.py
input = open(0).readline
MOD = 10**9 + 7

def pow(base: int, exp: int, mod: int):
    if exp == 1:
        return base
        
    next_base = pow(base, exp // 2, mod)
    if exp % 2 == 0:
        return next_base * next_base % mod
    else:
        return next_base * next_base * base % mod

for _ in range(T := int(input())):
    N = int(input())
    print(1 if N <= 2 else pow(2, N - 2, MOD))